ArduCopter/commands.cpp

```diff
@@ -21,34 +21,38 @@ void Copter::update_home_from_EKF()
 void Copter::set_home_to_current_location_inflight() {
     // get current location from EKF
     Location temp_loc;
-    if (inertial_nav.get_location(temp_loc)) {
-        const struct Location &ekf_origin = inertial_nav.get_origin();
-        temp_loc.alt = ekf_origin.alt;
-        if (!set_home(temp_loc, false)) {
-            return;
-        }
+    if (!ahrs.get_location(temp_loc)) {
+        return;
+    }
+    Location ekf_origin;
+    if (!ahrs.get_origin(ekf_origin)) {
+        return;
```

Comment:
Humm I wonder if we should really return here? I think that setting home to current altitude is still better than not setting it (I think this is impossible to happen by the way, if we have location we also have origin, but we should definitely check it).